### DISCUSSION TITLE
Export to XML workarounds?

### DISCUSSION THREAD

**Discussion by bdabelanger** [RevealBi/Reveal.Sdk#335] [General]
Hello!  I work with a long-time Reveal customer and we have deeply embedded Reveal BI tools into our PaaS

Our clients have asked if we can support .XML export of their report data, ideally similar to Excel export

I understand that this isn't achievable currently with Reveal export - I'm curious if there are any plans to add this functionality or if any other organizations have run into this issue and have found workarounds

We are temporarily recommending an Excel to XML converting tool/script which is proving to be a bit too fragile/finicky for our clients

**Comment by bdabelanger** [comment]
I see a relevant question here: https://github.com/RevealBi/Reveal.Sdk/discussions/62

I'm curious if something similar would be recommended here

**Comment by brianlagunas** [comment]
There are currently no plans to support an XML export. There have been no other requests for such an export. #62 does not apply to your request.

One thing you may be able to do is to get the raw data from the visualization and then generate your own custom xml export logic. We don't have a public API for this, but you could use a backdoor. You'd just need to know the index of the visualization you want the data for.

It would look something like this:

```ts
const data = revealView._dashboardView.__widgets[0].data()._table;
const json = data.toJson();
```

Once you have the data, you can create your own XML structure from the json.

Let me know if this helps.
